plugins/fileservices/fileservices.hpp

```diff
@@ -142,6 +142,7 @@ FILESERVICES_API void  FILESERVICES_CALL fsCreateExternalDirectory(ICodeContext
 FILESERVICES_API char * FILESERVICES_CALL fsfGetLogicalFileAttribute(ICodeContext * ctx,const char *lfn,const char *attrname);
 FILESERVICES_API void FILESERVICES_CALL fsProtectLogicalFile(ICodeContext * ctx,const char *lfn,bool set);
 FILESERVICES_API void FILESERVICES_CALL fsDfuPlusExec(ICodeContext * ctx,const char *_cmd);
+FILESERVICES_API const char * FILESERVICES_CALL fsGetEspURL(const char *username, const char *userPW);
```

Comment:
Wondering why return type here is const char * when fsfGetLogicalFileAttribute returns char * (both say varstring return type in the ecl definition). Which is right I wonder @ghalliday

Reply:
Richard is right - this should be a char *.  It is const in the sense that it shouldn't be modified, but non const in the sense that it can be freed.  The code generator creates the following prototype:

    extern "C"  char * fsGetEspURL(const char * username,const char * userpw);

Reply:
Changed return value from const char* to char*.